docker/ssl/access.js

```diff
@@ -0,0 +1,125 @@
+var clientKey = '';
+
+function access(s) {
+    s.on('upload', function (data) {
+        while (data == '') {
+            return s.AGAIN
+        }
+
+        if (clientKey === '') {
+            clientKey = parseCert(s.variables.ssl_client_s_dn, 'CN');
+        }
+
+        var pass = parsePackage(s, data);
+
+        if (!clientKey.length || pass !== clientKey) {
+            s.error('Cert CN (' + clientKey + ') does not match client ID');
+            s.off('upload')
+            s.deny();
+        }
+
+        s.off('upload');
+        s.allow();
+    })
+}
+
+function parsePackage(s, data) {
+    // An explanation of MQTT packet structure can be found here:
+    // https://public.dhe.ibm.com/software/dw/webservices/ws-mqtt/mqtt-v3r1.html#msg-format. 
+    var packet_type_flags_byte = data.codePointAt(0);
+    // First MQTT packet contain message type and flags. CONN message type
+    // is encoded as 0001, and we're not interested in flags, so all values
+    // 0001xxxx are valid for us, which is between 16 and 32.
```

Comment:
It is usual prctice to use ascii-art tables to depict protocol messages. For example you can take a look here: https://github.com/dustin/go-coap/blob/master/message.go#L444

Pictures like this can be found in IEEE RFCs.